Microsoft.Toolkit/Extensions/TaskExtensions.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Contracts;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.Toolkit.Extensions
+{
+    /// <summary>
+    /// Helpers for working with tasks.
+    /// </summary>
+    public static class TaskExtensions
```

Comment:
I'm not really sure they should be in the MVVM package though, as they're not really related to MVVM per se, and especially the generic version could be useful to other general .NET devs as well, which is why I've put these in the base `Microsoft.Toolkit` package. What I'm thinking here is that I wouldn't really expect people to **just** reference the MVVM package and not the base package too, especially so considering it's a dependency of most other toolkit packages as well. What's more, I was actually considering adding a reference to it in here too once #3346 gets merged, as it would simplify the code in here a bit more. And in that case, these extensions would be visible to anyone using the MVVM package as well. I've also structured them in a way that will work on the XAML compiler for `x:Bind` too (the non-generic one is above the generic one). This is all to say, I feel like leaving them here would be the best of both worlds for the largest number of potential users? 🤔

Reply:
I guess the question is if we reference the base package anyway, as then it's a bit of a moot point. I thought we originally decided we wanted this to be as independent as possible though, eh?

Reply:
That's true, I've just been reconsidering that lately especially after opening #3346. Basically:

- I don't like having these general-purpose extensions in the MVVM package, also because as I mentioned they'd reduce the available API surface for general .NET devs just using the base package.
- If those were just in the base package, it'd also make it easier for app devs only using the base package and not the MVVM one (eg. devs not using MVVM at all and just code behind, for instance) to write a converter or use them in their app without having to either copy-paste or pull the entire MVVM package just for that.
- Having as few dependencies as possible is important especially in performance critical scenarios (eg. the `HighPerformance` package is self contained). That's not really the case for the MVVM package, and the base toolkit package is self-contained anyway, and also extremely small.
- Switching to the new `ThrowHelper` APIs in the MVVM package would also simplify the code a bit in the MVVM package itself, as we could remove all the various throw-helper methods we have and just use those APIs.
- I don't think we'd have users potentially giving up on using the MVVM package just because it would have the single additional dependency to `Microsoft.Toolkit`.

Just thinking out loud here, let me know what you think! 😊